oracle/datadog_checks/oracle/oracle.py

```diff
@@ -216,14 +212,16 @@ def _get_custom_metrics(self, con, custom_queries, global_tags):
 
                     # Only submit metrics if there were absolutely no errors - all or nothing.
                     else:
+                        errored = False
                         for info in metric_info:
                             metric, value, method = info
                             getattr(self, method)(metric, value, tags=query_tags)
 
-    def _get_sys_metrics(self, con, tags):
-        if tags is None:
-            tags = []
+                    if errored:
+                        # If we failed to parse one row of the results, there is no reason to continue
+                        break
```

Comment:
Could/should we instead do something like https://github.com/DataDog/integrations-core/blob/dbec73a781cc8f812549222263192f7cc751ea4e/ibm_db2/datadog_checks/ibm_db2/ibm_db2.py#L409-L417

Reply:
Maybe I miss something, but isn't that done to catch SQL errors? We could do that as well, but the errored flag is here to catch parsing issues, not SQL errors.